warehouse/query-core/src/main/java/datawave/query/jexl/functions/IdentityAggregator.java

```diff
@@ -106,7 +106,8 @@ public Key apply(SortedKeyValueIterator<Key,Value> itr, Document doc, AttributeF
             for (Tuple2<String,String> fieldNameValue : fieldNameValues) {
                 Attribute<?> attr = attrs.create(fieldNameValue.first(), fieldNameValue.second(), topKey, true);
                 // only keep fields that are index only and pass the attribute filter
-                boolean toKeep = (fieldsToKeep == null || fieldsToKeep.contains(JexlASTHelper.removeGroupingContext(fieldNameValue.first())))
+                boolean toKeep = (fieldsToKeep == null || fieldsToKeep.isEmpty()
```

Comment:
The list of "indexOnlyFields" which includes the non-event fields is pased in as the fieldsToKeep from the QueryIterator.  If is is empty, then we should not be keeping any fields.  I believe this change is incorrect.